### PR TITLE
Drop the dependency on sphinx-testing

### LINKED ISSUE #8: 0.2.1: test suite requires sphinx_testing module which just has been archived
Yeah .. test suite requires sphinx_testing dodule which just has been archived and now is no longer maintained.
https://github.com/sphinx-doc/sphinx-testing/issues/21

### PR DESCRIPTION
**Pull request by hroncok**
Fixes https://github.com/MrSenko/sphinx-removed-in/issues/8

### CODE REVIEW

File: tests/test_extension.py [hunk 1]
```diff
@@ -1,22 +1,16 @@
...
-from sphinx_testing import with_app
+import pytest
 
 
-sys.path.insert(0,
-                os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
+PARENT = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..')
+sys.path.insert(0, PARENT)
 
+@pytest.mark.sphinx(buildername='html', srcdir=os.path.join(PARENT, 'docs'))
```
Comment (atodorov) [inline tests/test_extension.py:10]: I'm curious if this coming built-in with pytest or comes from the Sphinx dependency?

**Review by atodorov — CHANGES_REQUESTED** [review]
Overall looks good to me but we need a GitHub Actions CI job configured which will demonstrate that the new test is working.

File: tests/test_extension.py (hunk 1 above)
Comment (hroncok) [reply tests/test_extension.py:10]: From Sphinx.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1 @@
+pytest_plugins = 'sphinx.testing.fixtures'
```
Comment (hroncok) [inline tests/conftest.py:1]: @atodorov Here it needs to be enabled.

**Comment by hroncok** [comment]
I've just inherited the python-sphinx-testing package in Fedora and I've seen it is deprecated upstream, so I've looked at all the depending packages to see if we can remove that package from Fedora. This PR is a result of that. I provided it here considering you might benefit from it as well. However, I did not anticipate writing GtiHub Actions jobs. Is that really a requirement?

**Comment by atodorov** [comment]
> However, I did not anticipate writing GtiHub Actions jobs. Is that really a requirement?

Not really but I don't have enough time to pay proper attention to this repository at the moment. If you are in a hurry to get this change merged I need to see that the test suite still works. If you aren't then I will get back to it when I have the time.

**Comment by hroncok** [comment]
Done. It fails with Python 2. I can if/else it entirely to support Python 2 if that's something you want to keep supporting.

**Comment by hroncok** [comment]
No, I am not in any hurry. I planned to include this as a patch.

**Comment by hroncok** [comment]
It seems we are both force-pushing here. I've included a commit that restores Python 2 compatibility.

**Comment by hroncok** [comment]
I see what happened. Will rebase.

**Comment by hroncok** [comment]
Rebased.

**Comment by hroncok** [comment]
Thanks.

**Pull request merged**